plugins/eddn.py

```diff
@@ -1307,6 +1308,93 @@ def export_journal_fssbodysignals(
         this.eddn.export_journal_entry(cmdr, entry, msg)
         return None
 
+    def batch_journal_fsssignaldiscovered(
+        self, cmdr: str, system_name: str, system_starpos: list, is_beta: bool, entry: MutableMapping[str, Any]
+    ) -> Optional[str]:
+        """
+        Augment and keep FSSSignalDiscovered journal events
+
+        :param cmdr: the commander under which this upload is made
+        :param system_name: Name of current star system
+        :param system_starpos: Coordinates of current star system
+        :param is_beta: whether or not we are in beta mode
+        :param entry: the journal entry to batch
+        """
+        # {
+        #  "timestamp": "2021-01-15T02:54:18Z",
+        #  "event": "FSSSignalDiscovered",
+        #  "SystemAddress": 1900279744859,
+        #  "SignalName": "$USS_HighGradeEmissions;",
+        #  "SignalName_Localised": "Unidentified signal source",
+        #  "USSType": "$USS_Type_ValuableSalvage;",
+        #  "USSType_Localised": "Encoded emissions",
+        #  "SpawningState": "",
+        #  "SpawningFaction": "Free Marlinists of Carinae",
+        #  "ThreatLevel": 0,
+        #  "TimeRemaining": 718.508789
+        # }
+        #######################################################################
+        # Elisions
+        entry = filter_localised(entry)
+        if "USSType" in entry and entry["USSType"] == "$USS_Type_MissionTarget;":
+            logger.warning("USSType is $USS_Type_MissionTarget;, dropping")
+            return 'Dropping $USS_Type_MissionTarget;'
+        # Can check SystemAddress here, but we'll remove it from this signal list, to be added to the batch
+        if this.systemaddress is None or this.systemaddress != entry['SystemAddress']:
```

Comment:
Now this is a case for having the augmentations be *optional* (but that needs discussing over on https://github.com/EDCD/EDDN/pull/154 .

My thinking here is that it's better to have *all* the FSSSignalDiscovered events for a system, even if quirks of them mean we can't always have more than the `SystemAddress` for them.